apps/mobile/src/components/GalleryEditor/GalleryEditorRender.tsx

```diff
@@ -0,0 +1,138 @@
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';
+import { FlashList, ListRenderItem } from '@shopify/flash-list';
+import { useCallback, useEffect, useMemo } from 'react';
+import { NativeScrollEvent, NativeSyntheticEvent, View } from 'react-native';
+import { useAnimatedRef, useSharedValue } from 'react-native-reanimated';
+import { graphql, useFragment } from 'react-relay';
+
+import { useGalleryEditorActions } from '~/contexts/GalleryEditor/GalleryEditorContext';
+import { StagedSection } from '~/contexts/GalleryEditor/types';
+import { GalleryEditorHeaderFragment$key } from '~/generated/GalleryEditorHeaderFragment.graphql';
+import { GalleryEditorRenderFragment$key } from '~/generated/GalleryEditorRenderFragment.graphql';
+import { GalleryEditorRenderQueryFragment$key } from '~/generated/GalleryEditorRenderQueryFragment.graphql';
+import { GalleryEditorSectionFragment$key } from '~/generated/GalleryEditorSectionFragment.graphql';
+import { RootStackNavigatorParamList, RootStackNavigatorProp } from '~/navigation/types';
+
+import { useSafeAreaPadding } from '../SafeAreaViewWithPadding';
+import { GalleryEditorHeader } from './GalleryEditorHeader';
+import { GalleryEditorNavbar } from './GalleryEditorNavbar';
+import { GalleryEditorSection } from './GalleryEditorSection';
+
+export type ListItemType =
+  | { kind: 'navigation'; title: string }
+  | { kind: 'header'; galleryRef: GalleryEditorHeaderFragment$key }
+  | { kind: 'section'; section: StagedSection; queryRef: GalleryEditorSectionFragment$key };
+
+type Props = {
+  galleryRef: GalleryEditorRenderFragment$key;
+  queryRef: GalleryEditorRenderQueryFragment$key;
+};
+
+export function GalleryEditorRender({ galleryRef, queryRef }: Props) {
```

Comment:
this is freaking incredible

nit: i think we want to name this `GalleryEditorRenderer`